update doc for BoundServiceAccountTokenVolume (#27635)
kubernetes/website: content/en/docs/reference/access-authn-authz/service-accounts-admin.md

It acts synchronously to modify pods as they are created or updated. When this plugin is active
(and it is by default on most distributions), then it does the following when a pod is created or modified:

1. If the pod does not have a `ServiceAccount` set, it sets the `ServiceAccount` to `default`.
1. It ensures that the `ServiceAccount` referenced by the pod exists, and otherwise rejects it.
1. It adds a `volume` to the pod which contains a token for API access if neither the ServiceAccount `automountServiceAccountToken` nor the Pod's `automountServiceAccountToken` is set to `false`.
1. It adds a `volumeSource` to each container of the pod mounted at `/var/run/secrets/kubernetes.io/serviceaccount`, if the previous step has created a volume for ServiceAccount token.
1. If the pod does not contain any `ImagePullSecrets`, then `ImagePullSecrets` of the `ServiceAccount` are added to the pod.

#### Bound Service Account Token Volume

{{< feature-state for_k8s_version="v1.21" state="beta" >}}

When the `BoundServiceAccountTokenVolume` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/) is enabled, the service account admission controller will

> add the following projected volume instead of a secret volume for the non-expiring service account token created by Token Controller.

Reviewer: ```suggestion
add the following projected volume instead of a Secret-based volume for the non-expiring service account token created by Token Controller.
```

```yaml

> - name: kube-api-access-c5cs8

Reviewer: ```suggestion
- name: kube-api-access-<random-suffix>
```

  projected:
    defaultMode: 420 # 0644
    sources:
      - serviceAccountToken:
          expirationSeconds: 3600
          path: token
      - configMap:
          items:
            - key: ca.crt
              path: ca.crt
          name: kube-root-ca.crt
      - downwardAPI:
          items:
            - fieldRef:
                apiVersion: v1
                fieldPath: metadata.namespace
              path: namespace
```

This projected volume consists of three sources:

> 1. A ServiceAccountToken acquired from kube-apiserver via TokenRequest API. It will expire after 1 hour by default or the pod is deleted. It is bound to the pod and has kube-apiserver as the audience.

Reviewer: ```suggestion
1. A ServiceAccountToken acquired from kube-apiserver via TokenRequest API. It will expire after 1 hour by default or when the pod is deleted. It is bound to the pod and has kube-apiserver as the audience.
```

> 1. A ConfigMap contains a CA bundle used for verifying connections to the kube-apiserver. This feature depends on the `RootCAConfigMap` feature gate enabled which publish a "kube-root-ca.crt" ConfigMap to every namespace.

Reviewer: ```suggestion
1. A ConfigMap containing a CA bundle used for verifying connections to the kube-apiserver. This feature depends on the `RootCAConfigMap` feature gate being enabled, which publishes a "kube-root-ca.crt" ConfigMap to every namespace.
```

> 1. A ConfigMap containing a CA bundle used for verifying connections to the kube-apiserver. This feature depends on the `RootCAConfigMap` feature gate being enabled, which publishes a "kube-root-ca.crt" ConfigMap to every namespace. `RootCAConfigMap` is enabled by default in 1.20, and always enabled in 1.21+.

Reviewer 2: nit: I'm not sure I understand: 
`RootCAConfigMap` is enabled by default in 1.20, and always enabled in 1.21+?

Reviewer: it is on by default in 1.20 but can be disabled. it is unconditionally enabled in 1.21+

1. A DownwardAPI that references the namespace of the pod.

> See more details about [projected volume](/docs/tasks/configure-pod-container/configure-projected-volume-storage/).

Reviewer: ```suggestion
See more details about [projected volumes](/docs/tasks/configure-pod-container/configure-projected-volume-storage/).
```
